pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -112,9 +112,9 @@ func (state PipelineRunState) ToMap() map[string]*ResolvedPipelineTask {
 // IsBeforeFirstTaskRun returns true if the PipelineRun has not yet started its first TaskRun
 func (state PipelineRunState) IsBeforeFirstTaskRun() bool {
 	for _, t := range state {
-		if t.IsCustomTask() && t.RunObject != nil {
+		if t.IsCustomTask() && len(t.RunObjects) > 0 {
```

Comment:
This makes me think this function did not work correctly for matrixed runs. Can you check test coverage here?

I think this can also be simplified to
```
for _, t := range state {
  if len(t.RunObjects) > 0 || len(t.TaskRuns) > 0 {
    return false
  }
}
```

Reply:
Updated and added test coverage for matrixed runs.